force-app/main/default/triggers/StudentTrigger.trigger

```diff
@@ -0,0 +1,14 @@
+trigger StudentTrigger on Student__c (before insert) {
+    if(trigger.isInsert){
+        if(trigger.isBefore){
+            StudentManager.beforeInsertStudent(Trigger.New);
+        }else if(trigger.isAfter){
+            StudentManager.afterUpdateInsertStudent(Trigger.New,null);
+        }
+    }else if(trigger.isUpdate){
```

Comment:
Since this trigger only working for `before insert` event then how this condition going to work?